File: webflux-dialogue/src/test/java/com/study/webflux/rag/config/annotation/ControllerWebFluxTest.java
package com.study.webflux.rag.config.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
import org.springframework.core.annotation.AliasFor;

import org.junit.jupiter.api.Tag;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@WebFluxTest
@Tag("controller")
public @interface ControllerWebFluxTest {

	@AliasFor(annotation = WebFluxTest.class, attribute = "controllers")
	Class<?>[] value() default {};

}

File: backend/bootstrap/src/test/kotlin/com/miyou/app/config/annotation/ControllerWebFluxTest.kt
package com.miyou.app.config.annotation

import org.junit.jupiter.api.Tag
import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest
import org.springframework.core.annotation.AliasFor
import kotlin.reflect.KClass

/**
 * WebFlux 컨트롤러 테스트 시 중복 설정을 방지하고 일관된 태깅을 적용하기 위한 커스텀 어노테이션.
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@WebFluxTest
@Tag("controller")
annotation class ControllerWebFluxTest(
    @get:AliasFor(annotation = WebFluxTest::class, attribute = "controllers")
    val value: Array<KClass<*>> = [],
)
